core/src/main/java/de/jplag/Submission.java

```diff
@@ -284,11 +292,14 @@ private static File createErrorDirectory(String... subdirectoryNames) {
     private void extractAndParseComments() {
         Optional<CommentExtractorSettings> commentExtractorSettings = language.getCommentExtractorSettings();
         if (commentExtractorSettings.isPresent()) {
+            List<Comment> rawComments = new ArrayList<>();
             for (File file : files) {
                 CommentExtractor extractor = new CommentExtractor(file, commentExtractorSettings.get());
-                comments.addAll(extractor.extract());
+                rawComments.addAll(extractor.extract());
```

Comment:
Also here, the `CommentExtractor` is a throwaway object. Can we initialize it once with the settings and use it multiple times?